Upgraded Plone and requirements

pnicolli opened the pull request:
A couple upgrades for the upcoming Ploneconf Trainings in Tokyo:

- Plone 5.1.4
- setuptools 39.1.0
- zc.buildout 2.11.4

setuptools and zc.buildout might not be the latest already, but are the newest that I've tried and are working properly, feel free to provide feedback if you tried some newer versions.

pbauer commented:
+1 the version need to be in sync with http://dist.plone.org/release/5.1.4/versions.cfg so this is ok.

pnicolli commented:
@pbauer I added a small commit to fix travis at the same time you merged and deleted, so github actually reopened the branch with the new commit... you can just delete the `plone514` branch again :)
